jubilant/_juju.py

```diff
@@ -563,7 +563,7 @@ def destroy_model(
             self.model = None
 
     @overload
-    def exec(self, command: str, *args: str, machine: int, wait: float | None = None) -> Task: ...
+    def exec(self, command: str, *args: str, machine: str, wait: float | None = None) -> Task: ...
```

Comment:
@benhoyt Dima, James, and I were discussing this and undecided whether this needs to be `int | str` to be backwards compatible - so static type checking of people's Jubilant code doesn't break from this change (forcing them to put in a `str()` or `''`).

We could accept `int | str`, since we just `str()` it ourselves. But it seems cleaner to me to be `str` since that's the correct type. I feel like breaking compatibility is ok for a bug fix - the charm test code is technically wrong if it's using an int, so it should actually get fixed.

Your thoughts?